scripts/west_commands/runners/qemu.py

```diff
@@ -4,28 +4,210 @@
 
 '''Runner stub for QEMU.'''
 
-from runners.core import ZephyrBinaryRunner, RunnerCaps
+import argparse
+import logging
+import os
+from pathlib import Path
+import shlex
+import subprocess
+import sys
+import yaml
 
+from runners.core import ZephyrBinaryRunner, RunnerCaps, FileType
+
+DEFAULT_QEMU_EXE = 'qemu-system-arm_rt595.exe' if sys.platform == 'win32' else 'qemu-system-arm_rt595'
+DEFAULT_QEMU_GDB_PORT = 1234
+DEFAULT_QEMU_FIFO = None
 
 class QemuBinaryRunner(ZephyrBinaryRunner):
     '''Place-holder for QEMU runner customizations.'''
 
+    def __init__(self, cfg, machine, cpu, dtb, qemu_opts,
+                 commander=DEFAULT_QEMU_EXE,
+                 pipe=DEFAULT_QEMU_FIFO,
+                 gdbserver='',
+                 gdb_host='127.0.0.1',
+                 gdb_port=DEFAULT_QEMU_GDB_PORT,
+                 tui=False, tool_opt=[]):
+        super().__init__(cfg)
+        self.file = cfg.file
+        self.file_type = cfg.file_type
+        self.hex_name = cfg.hex_file
+        self.bin_name = cfg.bin_file
+        self.elf_name = cfg.elf_file
+        self.gdb_cmd = [cfg.gdb] if cfg.gdb else None
```

Comment:
Just FYI, you can access all of these using `self.cfg.hex_file`, etc.

There is no need to save them in extra variables.

Reply:
thanks, not check the parent class for this. let me update